### PR TITLE
Improve form key

### PR DESCRIPTION
**Pull request by ingorichtsmeier**
First draft to prefer `formRef` over `formKey` for Camunda Forms

### CODE REVIEW

**Review by smbea — COMMENTED** [review]
I think the explanation is simple and clear, and it motivates users to use formRef as default. And it goes in line with https://github.com/bpmn-io/bpmn-js-properties-panel/pull/574. WDYT @MaxTru ?

**Comment by MaxTru** [comment]
Makes sense!

File: content/user-guide/task-forms/_index.md [hunk 1]
```diff
@@ -99,20 +99,9 @@ Users can still submit any variables via APIs for form completion like the
...
-* `camunda-forms:app:forms/FORM_NAME.form`
-
-When you use [Camunda Platform Run](../camunda-bpm-run) we recommend the camunda-forms:deployment form key for easy deployment from the Camunda Modeler.
-
-{{< img src="img/reference-camunda-form.png" title="Provide Form Key for Camunda Forms" >}}
-
 ## Form Reference
 
-With Form References, Camunda Forms provide an additional way of linking an element in a BPMN diagram to a form. To link a BPMN element ([StartEvent][start-event] or [UserTask][user-tasks]) to a Camunda Form, you have to specify the key of the Camunda Form as the `camunda:FormRef` attribute. Additionally, the `camunda:formRefBinding` attribute specifies which version of the Camunda Form to reference.
+With Form References, Camunda Forms provide a flexible way of linking an element in a BPMN diagram to a form. To link a BPMN element ([StartEvent][start-event] or [UserTask][user-tasks]) to a Camunda Form, you have to specify the Id of the Camunda Form as the `camunda:FormRef` attribute. Additionally, the `camunda:formRefBinding` attribute specifies which version of the Camunda Form to reference.
```
Comment (MaxTru) [inline content/user-guide/task-forms/_index.md:104]:
```suggestion
With Form References, Camunda Forms provide a flexible way of linking an element in a BPMN diagram to a form. To link a BPMN element ([StartEvent][start-event] or [UserTask][user-tasks]) to a Camunda Form, you have to specify the Id of the Camunda Form as the `camunda:formRef` attribute. Additionally, the `camunda:formRefBinding` attribute specifies which version of the Camunda Form to reference.
```

**Review by MaxTru — CHANGES_REQUESTED** [review]
Just one small adjustment, then "ready to merge" from my perspective

**Review by MaxTru — APPROVED** [review]

**Pull request merged**